theano/sandbox/rng_mrg.py

```diff
@@ -887,15 +887,18 @@ def binomial(self, size, n=1, p=0.5,
             A Theano tensor of samples randomly drawn from a binomial distribution.
 
         """
-        # TODO: use ABCs for these checks?
-        if not isinstance(p, theano.Variable):
+        if not isinstance(n, theano.Variable):
+            assert n == int(n), "n must be an integer"
             if n <= 0:
-                raise ValueError("Number of experiments can not be negative or zero for a binomial distribution", n)
+                raise ValueError("n can not be negative or zero", n)
             elif n > 10:
-                warnings.warn("Current implementation binomial distribution is memory-inefficient for large n (%d)" % n)
+                warnings.warn("Current implementation binomial distribution leads to "
+                              "memory consumption linear in n (%d)" % n)
 
-        if not isinstance(p, theano.Variable) and (p < 0. or p > 1.):
-            raise ValueError("Probability of success must be between 0 and 1 for a binomial distribution", p)
+        if not isinstance(p, theano.Variable):
+            assert p == float(p), "p must be a float"
+            if not 0. < p < 1.:
```

Comment:
p can be 0 or 1, so it should be `if not 0. <= p <= 1.` instead !

Reply:
I would like to apologise for my idiocy...